saltychat/SaltyClient/VoiceManager.cs

```diff
@@ -614,6 +660,16 @@ private async Task OnControlTick()
                     this.ToggleVoiceRange();
                 }
 
+                if (Game.PlayerPed.IsInPoliceVehicle) {
+                    if (API.GetPedInVehicleSeat(Game.PlayerPed.CurrentVehicle.Handle, -1) != 0 || API.GetPedInVehicleSeat(Game.PlayerPed.CurrentVehicle.Handle, 0) != 0)
+                    {
+                        if (Game.IsControlJustPressed(0, Control.SpecialAbilitySecondary))
+                            BaseScript.TriggerServerEvent(Event.SaltyChat_IsUsingMegaphone, true);
+                        else if (Game.IsControlJustReleased(0, Control.SpecialAbilitySecondary))
+                            BaseScript.TriggerServerEvent(Event.SaltyChat_IsUsingMegaphone, false);
+                    }
+                }
```

Comment:
This should be further optimized, so it doesn't get 3 times the own ped and 2 time the vehicle we are in.
You could also optimize it more by checking if the key is pressed/released first and then check all the other stuff, but eeehh... who cares.
You also built yourself a trap and should add a tracker (aka a private field):
- Enter a police vehicle
- Hold down the keybind
- While still holding it, leave the vehicle
- Can't stop talking on the megaphone, until you get back in the car and press/release the keybind

```suggestion
                Ped playerPed = Game.PlayerPed;

                if (playerPed.IsInPoliceVehicle)
                {
                    Vehicle vehicle = playerPed.CurrentVehicle;

                    if (vehicle.GetPedOnSeat(VehicleSeat.Driver) == playerPed || vehicle.GetPedOnSeat(VehicleSeat.Passenger) == playerPed)
                    {
                        if (Game.IsControlJustPressed(0, Control.SpecialAbilitySecondary))
                        {
                            BaseScript.TriggerServerEvent(Event.SaltyChat_IsUsingMegaphone, true);

                            // set tracker to `true`
                        }
                        else if (Game.IsControlJustReleased(0, Control.SpecialAbilitySecondary))
                        {
                            BaseScript.TriggerServerEvent(Event.SaltyChat_IsUsingMegaphone, false);

                            // set tracker to `false`
                        }
                    }
                }
                // else if (tracker)
                //      BaseScript.TriggerServerEvent(Event.SaltyChat_IsUsingMegaphone, false);
                //      set tracker to `false`
```

Reply:
Yeah oversaw that trap, but I further optimized the code. Should be better now